Build/build.ps1

```diff
@@ -15,22 +15,158 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+
+.DESCRIPTION
+Constructs a query using the current commit 'sha' to retrieve the Pull Request information
+via the GitHub API using the 'Auth Token' and returns the description of the found pull request.
+The 'Auth Token' and current commit 'sha' are available through Travis CI environment variables.
+#>
+function Get-PullRequestDescription {
+    [CmdletBinding()]
+    [OutputType([string])]
+
+    $searchIssuesAPIUrl = 'https://api.github.com/search/issues'
+    $repository = 'vmware/dscr-for-vmware'
+    $searchType = 'pr'
+    $pullRequestState = 'closed'
+
+    $base64Token = [System.Convert]::ToBase64String([char[]] $env:GH_TOKEN)
+    $uri = "$searchIssuesAPIUrl?q=repo:$repository+type:$searchType+state:$pullRequestState+$env:TRAVIS_COMMIT is:merged"
+    $method = 'Get'
+    $headers = @{
+        Authorization = "Basic $base64Token"
+    }
+
+    try {
+        $pullRequestsInfo = Invoke-RestMethod -Uri $uri -Method $method -Headers $headers
+    }
+    catch {
+        throw "An error occurred while retrieving the Pull Request information: $($_.Exception.Message)"
+    }
+
+    $pullRequest = $pullRequestsInfo.items[0]
+    if ($null -eq $pullRequest) {
+        <#
+        Here the build was triggered without a Pull Request so the description should be
+        retrieved from the commit body.
+        #>
+        $commitUri = "https://api.github.com/repos/$repository/commits/$env:TRAVIS_COMMIT"
+        try {
+            $commitInfo = Invoke-RestMethod -Uri $commitUri -Method $method -Headers $headers
+        }
+        catch {
+            throw "An error occurred while retrieving the Commit information: $($_.Exception.Message)"
+        }
+
+        return $commitInfo.commit.message
+    }
+
+    return $pullRequest.body
+}
+
+<#
+.SYNOPSIS
+Updates the Changelog document with the changes mentioned in the Pull Request description.
+
+.DESCRIPTION
+Updates the content of the Changelog document adding a new section containing
+the new module version and the changes made with the Pull Request which are mentioned in the description.
+
+.PARAMETER ChangelogDocumentPath
+Specifies the path to the Changelog document which is going to be updated.
+
+.PARAMETER PullRequestDescription
+Specifies the description of the changes made in the Pull Request.
+
+.PARAMETER ModuleVersion
+Specifies the new Module Version that is going to be added to the new section in the document.
+#>
+function Update-ChangelogDocument {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory = $true)]
+        [string] $ChangelogDocumentPath,
+
+        [Parameter(Mandatory = $true)]
+        [string] $PullRequestDescription,
+
+        [Parameter(Mandatory = $true)]
+        [string] $ModuleVersion
+    )
+
+    $changelogDocument = Get-Content -Path $ChangelogDocumentPath
+
+    # The Header in the CHANGELOG.md consists of the first two lines plus one empty line so we need the first three lines of the file.
+    $headerLength = 2
```

Comment:
Hm, hardcoded number of lines for the header is tricky. More relieble will be to search for the top changelog record and insert the new records on top of it.

Reply:
Done.